### PR TITLE
D2K - Subfaction Images and Voices

### PR DESCRIPTION
**Pull request by MustaphaTR**
Before this PR human player that uses subfactions get no unit or mentat voice. Now they use the voices from the same faction they share structure artwork with.

This fixes that Fremen uses Harkonnen structure artwork instead of Atreides.

This changes `Image:` tag for structures from Harkonnen ones to Ordos, as there are 3 Ordos and 2 other subfactions. Harkonnen and Corrino are now defined on `FactionImages:`, while Ordos, Mercenary and Smuggler are removed from there.

At last this replaces missing cameos with Ordos counterparts for Mercenary War Factory and Smuggler Starport (Those use Ornithopter Cameo in original game and was Atreides Combat Tank on ORA).

~~Depends on #14457, as you can't test the second commit without what that PR fixes.~~

### CODE REVIEW

**Comment by ltem** [comment]
I took a quick look over it and I couldn't spot any issues but I have to check it again in more detail, I should be done in the next days. However, I have two questions.


First, could it be that `Fremen` use the same palace-shp-file as `Corrino`?
I found this image some time ago and used it as a reference (I think I got it from gruntsmods). While the `Smuggler Starport` and `Mercenary Heavy Factory` are fine, the `Corrino` palace shape is used for the `Fremen`.

|structures|
|:---:|


Second, did you ever stumble over "crushing screams" in d2k, which gets played if infantry gets crushed by a vehicle? I'm asking because you're handling the sound files in this PR.

**Comment by MustaphaTR** [comment]
Palace is wierd, when you preplace using Map Editor, Fremen gets Emparor's Palace. Play as Fremen ingame and progress thru, you get Atreides Palace buildable. I'm not sure what is the case with Corrino itself, i may need to test.

**Comment by MustaphaTR** [comment]
Rebased.

File: mods/d2k/rules/structures.yaml [hunk 1]
```diff
@@ -568,12 +569,12 @@ outpost:
...
 		RequiresCondition: !disabled
 	RenderSprites:
-		Image: outpost.harkonnen
+		Image: outpost.ordos
 		FactionImages:
 			atreides: outpost.atreides
-			ordos: outpost.ordos
-			smuggler: outpost.ordos
-			mercenary: outpost.ordos
+			fremen: outpost.fremen
```
Comment (ltem) [inline mods/d2k/rules/structures.yaml]: Quite sure it is a typo `outpost.fremen` -> `outpost.atreides`

File: mods/d2k/sequences/structures.yaml [hunk 2]
```diff
@@ -1556,8 +1556,8 @@ starport.smuggler:
 		Frames: 643, 644, 645, 663, 664, 665
 		Length: 6
 		Offset: -16,-16
-	icon: DATA.R8 # TODO: blank
-		Start: 4281
+	icon: DATA.R8 # TODO This is same as Ordos one.
```
Comment (ltem) [inline mods/d2k/sequences/structures.yaml]: If there is no reason to keep the `TODO`, you could remove it.

File: mods/d2k/sequences/structures.yaml [hunk 3]
```diff
@@ -1606,8 +1606,8 @@ heavy.mercenary:
...
```
Comment (ltem) [inline mods/d2k/sequences/structures.yaml]: Same as above.

File: mods/d2k/rules/structures.yaml [hunk 4]
```diff
@@ -1038,13 +1040,12 @@ palace:
...
 		Range: 4c768
 	RenderSprites:
-		Image: palace.harkonnen
+		Image: palace.ordos
 		FactionImages:
 			atreides: palace.atreides
-			ordos: palace.ordos
+			fremen: palace.atreides
+			harkonnen: palace.harkonnen
 			corrino: palace.corrino
```
Comment (ltem) [inline mods/d2k/rules/structures.yaml]: The `Corrino palace` exit point is a bit off, could this be updated in this PR? IMO it isn't urgent.

**Review by ltem — CHANGES_REQUESTED** [review]

File: mods/d2k/rules/structures.yaml (hunk 4 above)
Comment (MustaphaTR) [reply mods/d2k/rules/structures.yaml]: Corrino palace isn't meant to be able to build stuff tbh. Considering #14460 maybe we should seperate the palaces.

File: mods/d2k/sequences/structures.yaml (hunk 2 above)
Comment (MustaphaTR) [reply mods/d2k/sequences/structures.yaml]: We have a custom artwork for corrino palace, maybe someone may want to make one for Smuggler Starport and Mercenary HFac.

**Comment by MustaphaTR** [comment]
Updated.

**Review by ltem — APPROVED** [review]

File: mods/d2k/rules/structures.yaml (hunk 4 above)
Comment (ltem) [reply mods/d2k/rules/structures.yaml]: Sounds good, I left a line about it at the issue.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/505762b096cfa5b5c4a68979e9a9c62125e56c7d

**Pull request merged**